src/components/Modal/spec.js

```diff
@@ -1,22 +1,86 @@
-/* globals expect, test */
+/* globals expect, describe, beforeAll, afterAll, it */
+import { mount } from 'enzyme'
 import React from 'react'
 import ReactDOM from 'react-dom'
 import renderer from 'react-test-renderer'
+import sinon from 'sinon'
 import { Modal, Provider } from '../../../dist'
 
-// react-test-renderer does not play nicely with ReactDOM.createPortal(), which
-// is used by Grommet's Layer component in the Modal component. To get it to
-// work without throwing an error, the ReactDOM.createPortal() method is stubbed
-// here.
-// see: https://github.com/facebook/react/issues/11565
-ReactDOM.createPortal = node => node
-
-test('Modal renders correctly', () => {
-  const component = renderer.create(
-    <Provider>
-      <Modal>Modal</Modal>
-    </Provider>
-  )
-  let tree = component.toJSON()
-  expect(tree).toMatchSnapshot()
+describe('Modal', () => {
+  let originalPortal
+  beforeAll(() => {
+    // react-test-renderer does not play nicely with ReactDOM.createPortal(), which
+    // is used by Grommet's Layer component in the Modal component. To get it to
+    // work without throwing an error, the ReactDOM.createPortal() method is stubbed
+    // here.
+    // see: https://github.com/facebook/react/issues/11565
+    originalPortal = ReactDOM.createPortal
+    ReactDOM.createPortal = node => node
+  })
+
+  afterAll(() => {
+    ReactDOM.createPortal = originalPortal
+  })
+
+  it('renders correctly', () => {
+    const component = renderer.create(
+      <Provider>
+        <Modal>Modal</Modal>
+      </Provider>
+    )
+    let tree = component.toJSON()
+    expect(tree).toMatchSnapshot()
+  })
+})
+
+describe('Nested modals', () => {
+  it('should pop the top-most modal on Escape key press', () => {
+    const firstModalSpy = sinon.spy()
+    const secondModalSpy = sinon.spy()
+
+    const component = mount(
+      <Provider>
+        <Modal done={firstModalSpy}>
+          <Modal done={secondModalSpy}>Test</Modal>
+        </Modal>
+      </Provider>
+    )
+
+    // Escape key
+    component
+      .find('div[onKeyDown]')
+      .forEach(node => node.simulate('keydown', { which: 27 }))
+    expect(firstModalSpy.notCalled).toBeTruthy()
+    expect(secondModalSpy.calledOnce).toBeTruthy()
+  })
+
+  it('should pop the top-most modal with done on Enter key press', () => {
+    const firstModalSpy = sinon.spy()
+    const secondModalSpy = sinon.spy()
+
+    // You cannot simulate events on the document object in enzyme, so we need to stub the `addEventListener` function.
+    const map = {}
+    window.document.addEventListener = (event, cb) => {
```

Comment:
Let's add a `describe()` and move this in a `before()` call and properly restore it in an `after()` call.